### PR TITLE
4806-document-different-ways-to-execute-queries-against-iox

### PR DESCRIPTION
**Pull request by jstirnaman**
Part of #4806

- Revise Grafana instructions.
- Add query section.
- Fix formatting.
- Add Homebrew help.
- DRY

I intended to follow this PR with another that moves everything except "Build visualizations..." into /query-data/execute-queries/flight-sql/grafana, i.e. my treatment of Superset. Now, I'm questioning whether we need both "Query data" and "Visualize data". I imagine we'll still want to provide more Visualize instruction, especially how to aggregate and pivot with queries for visualization purposes, but do we still need a separate section for that when most of it (AFAIK) will be done using external tools?
- [ ] Signed the [InfluxData CLA](https://www.influxdata.com/legal/cla/)
  ([if necessary](https://github.com/influxdata/docs-v2/blob/master/CONTRIBUTING.md#sign-the-influxdata-cla))
- [ ] Rebased/mergeable

### CODE REVIEW

File: layouts/shortcodes/sql/sql-schema-intro.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+When working with InfluxDB's implementation of SQL, a **bucket** is equivalent
```
Comment (sanderson) [inline layouts/shortcodes/sql/sql-schema-intro.md:1]:
```suggestion
When working with the InfluxDB SQL implementation, a **bucket** is equivalent
```

File: content/influxdb/cloud-iox/visualize-data/grafana.md [hunk 2]
```diff
@@ -24,40 +25,62 @@ InfluxDB bucket backed by InfluxDB IOx.
 >
 > {{% caption %}}[Grafana documentation](https://grafana.com/docs/grafana/latest/introduction/){{% /caption %}}
 
-For the most performant queries, use SQL and the
-[Flight SQL protocol](https://arrow.apache.org/blog/2022/02/16/introducing-arrow-flight-sql/)
-to query InfluxDB.
+<!-- TOC -->
 
+- [Set up Grafana](#set-up-grafana)
```
Comment (sanderson) [inline content/influxdb/cloud-iox/visualize-data/grafana.md:30]:
```suggestion
- [Install Grafana](#install-grafana)
```

File: content/influxdb/cloud-iox/visualize-data/grafana.md [hunk 3]
```diff
@@ -24,40 +25,62 @@ InfluxDB bucket backed by InfluxDB IOx.
...
+<!-- /TOC -->
+
+## Set up Grafana
+
+Follow the [Grafana.com instructions to Install Grafana](https://grafana.com/docs/grafana/latest/setup-grafana/installation/) for your system.
+
+{{% warn %}}
+Because Grafana Flight SQL Plugin is a custom plugin, you can't use it with Grafana Cloud.
+For more information, see [Find and Use Plugins in the Grafana Cloud documentation](https://grafana.com/docs/grafana-cloud/fundamentals/find-and-use-plugins/)
+{{% /warn %}}
```
Comment (sanderson) [inline content/influxdb/cloud-iox/visualize-data/grafana.md:49]: Move this down under "Download the Grafana Flight SQL plungin."

```suggestion
```

File: content/influxdb/cloud-iox/visualize-data/grafana.md [hunk 4]
```diff
@@ -24,40 +25,62 @@ InfluxDB bucket backed by InfluxDB IOx.
...
+
+## Download the Grafana Flight SQL Plugin
```
Comment (sanderson) [inline content/influxdb/cloud-iox/visualize-data/grafana.md:51]:
```suggestion
## Download the Grafana Flight SQL plugin
```

**Review by sanderson — CHANGES_REQUESTED** [review]
Looks really good. Just some minor suggestions.

File: content/influxdb/cloud-iox/visualize-data/grafana.md (hunk 3 above)
Comment (jstirnaman) [reply content/influxdb/cloud-iox/visualize-data/grafana.md:49]: > Move this down under "Download the Grafana Flight SQL plungin."

Thanks. Couldn't decide whether to keep both.

File: content/influxdb/cloud-iox/visualize-data/grafana.md [hunk 5]
```diff
@@ -202,18 +217,44 @@ to explicitly load unsigned plugins.
...
+1. Click **Explore**.
+2. In the dropdown, select the saved datasource that you want to query.
+3. Use the SQL query form to build your query:
+    - **FROM**: Select the measurement that you want to query.
+    - **SELECT**: Select one or more fields and tags to return as columns in query results.
+                  In Grafana, you must specify a **time** column in the `SELECT` list.
+    - **WHERE**: To filter the query results, enter a conditional expression.
+    - **GROUP BY**: To `GROUP BY` one or more fields or tags, enter them as a comma-delimited list.
+                    If you include an aggregate function in the **SELECT** list,
+                    then you must include a time value in a `GROUP BY` or `PARTITION BY` clause.
```
[the remark this answers is not in the document]
Comment (jstirnaman) [reply content/influxdb/cloud-iox/visualize-data/grafana.md:252]: > You can group by columns other than time.

True, but Grafana specifically requires a `time` column in the result and Flight SQL (or IOx?) doesn't know how to resolve `iox.[measurement].time` unless you group or partition by time. At least, as far as I can tell.

**Pull request merged**